### ISSUE TITLE
[BUG] : [Ministral-3-14B-Instruct-2512] expand() produces incorrect values — stride-0 dim materialized as raw memory copy

### ISSUE THREAD

**Issue by abhishekkunuru6-cmyk** [torch-spyre/torch-spyre#1470]
### What happened?

When calling .expand() on a tensor residing on the Spyre device and then transferring it back to CPU via .cpu(), the resulting tensor contains incorrect values and wrong strides.
The expanded tensor on-device correctly shows stride 0 on the expanded dimension — matching expected PyTorch behavior. However, after .cpu(), the strides change from (4, 1, 0) to (12, 3, 1), indicating the tensor was materialized as if contiguous during the device-to-host copy.
This causes the copy to read beyond the actual allocated storage (only 8 bytes exist for the original [1, 4, 1] tensor, but 24 bytes are read), producing garbage values in the output.
The same operation on CPU produces correct results. The bug is isolated to the Spyre→CPU transfer path for non-contiguous tensors created via expand().

**Example :**
```
src shape: [1, 4, 1]
Memory layout (8 bytes total, 4 values):
[ 1.00, -1.37,  1.39, -1.85 ]

after expand shape:[1,4,3]

Logical view [1, 4, 3] with strides (4, 1, 0):

        col0    col1    col2
row0  [  1.00,   1.00,   1.00  ]   ← all point to v0
row1  [ -1.37,  -1.37,  -1.37  ]   ← all point to v1
row2  [  1.39,   1.39,   1.39  ]   ← all point to v2
row3  [ -1.85,  -1.85,  -1.85  ]   ← all point to v3

Still only 8 bytes in memory. No new data allocated.

What we got as output with strides [12, 3, 1]:

Spyre expanded            :
 tensor([[ 1.0000e+00, -1.3691e+00,  1.3926e+00],
        [-1.8457e+00,  3.5059e+00,  2.5234e+01],
        [ 5.0625e+01,  1.7234e+01,  7.0953e-04],
        [-4.2531e+01,  1.0112e+02,  0.0000e+00]],

```

### What did you expect to happen?

When calling .expand() followed by .cpu() on a Spyre tensor, the resulting CPU tensor should contain the same values as the equivalent CPU operation.
Specifically,  each expanded column should repeat the original value, not read sequential memory. The expected output for a [1, 4, 1] tensor expanded to [1, 4, 3] is:
[  1.00,   1.00,   1.00 ]
[ -1.37,  -1.37,  -1.37 ]
[  1.39,   1.39,   1.39 ]
[ -1.85,  -1.85,  -1.85 ]

### How can we reproduce it?

Just run this code : 

```
import torch
import torch_spyre

DEVICE = "spyre"

src = torch.randn(1, 4, 1, dtype=torch.float16)

spyre_tensor = src.to(DEVICE).expand(1, 4, 3)
spyre_out    = spyre_tensor.cpu()


cpu_out = src.expand(1, 4, 3)


print("=== Strides ===")
print(f"Spyre strides (after .cpu()) : {spyre_out.stride()}")   
print(f"CPU   strides                : {cpu_out.stride()}")    

print("\n=== Values ===")
print(f"src            : {src.squeeze()}")
print(f"CPU  expanded  : {cpu_out.squeeze()}")
print(f"Spyre expanded : {spyre_out.squeeze()}")


print("\n=== Slice equality (expanded cols should all be equal) ===")
print(f"CPU   col0 == col1 : {torch.equal(cpu_out[:,:,0],   cpu_out[:,:,1])}")    
print(f"CPU   col0 == col2 : {torch.equal(cpu_out[:,:,0],   cpu_out[:,:,2])}")    
print(f"Spyre col0 == col1 : {torch.equal(spyre_out[:,:,0], spyre_out[:,:,1])}")  
print(f"Spyre col0 == col2 : {torch.equal(spyre_out[:,:,0], spyre_out[:,:,2])}")  

print("\n=== Tensor equality (cpu_out vs spyre_out) ===")
print(f"torch.equal : {torch.equal(cpu_out, spyre_out)}")                          
print(f"allclose    : {torch.allclose(cpu_out.float(), spyre_out.float())}")        
```

### Any environmental details we need to know?

Ensure the tests are executed within the torch-spyre environment.

### Anything else we need to know?

On-device behavior is correct:
The strides on Spyre before .cpu() are (4, 1, 0) — matching CPU exactly.
The bug only manifests during the transfer, not during the expand operation itself.

### Relevant log output

```Shell
(torch-spyre) [vishwasr@vishwasr-spyre-dev-pf model_centric_op_level_tests]$ python tets.py
=== Strides ===
Spyre strides (after .cpu()) : (12, 3, 1)
CPU   strides                : (4, 1, 0)

=== Values ===
src            : tensor([-1.2861,  0.6987, -1.7764,  1.7969], dtype=torch.float16)
CPU  expanded  : tensor([[-1.2861, -1.2861, -1.2861],
        [ 0.6987,  0.6987,  0.6987],
        [-1.7764, -1.7764, -1.7764],
        [ 1.7969,  1.7969,  1.7969]], dtype=torch.float16)
Spyre expanded : tensor([[-1.2852e+00,  6.9824e-01, -1.7754e+00],
        [ 1.7969e+00,  3.3600e+03,  3.2280e+03],
        [ 9.3312e+01,  0.0000e+00,  3.4304e+04],
        [ 3.0398e-06,  5.0562e+01,  5.0469e+01]], dtype=torch.float16)

=== Slice equality (expanded cols should all be equal) ===
CPU   col0 == col1 : True
CPU   col0 == col2 : True
Spyre col0 == col1 : False
Spyre col0 == col2 : False

=== Tensor equality (cpu_out vs spyre_out) ===
torch.equal : False
allclose    : False
```

**Comment by abhishekkunuru6-cmyk** [comment]
cc: @ani300 @saurabh-srivastava4 @goutham-binnadi-gopala

**Comment by tardieu** [comment]
@AdnanHoque is looking into it.

**Comment by tardieu** [comment]
Adding a related test case where the unsqueeze is happening on device:
```import torch

DEVICE = torch.device("spyre")


torch.manual_seed(0xAFFE)

x = torch.randn(5, 256, dtype=torch.float16)


def f(x):
    return (x).unsqueeze(0).expand(3, 5, 256)


cpu_result = f(x)
aiu_result = torch.compile(f)(x.to(DEVICE)).cpu()

print(torch.abs(aiu_result - cpu_result).amax())
```
